cljfmt/src/cljfmt/diff.clj

```diff
@@ -9,13 +10,18 @@
 (defn unlines [ss]
   (str/join "\n" ss))
 
+(defn to-absolute-path [filename]
+  (->> (str/split filename (re-pattern File/separator))
+       (apply io/file)
+       .getCanonicalPath))
+
 (defn unified-diff
   ([filename original revised]
    (unified-diff filename original revised 3))
   ([filename original revised context]
    (unlines (DiffUtils/generateUnifiedDiff
-             (str (io/file "a" filename))
-             (str (io/file "b" filename))
+             (->> filename to-absolute-path (str "a"))
```

Comment:
Example cases fixed with this:

* `lein cljfmt check ../../some-dir`
* `lein cljfmt check ~/some-dir`